src/main/resources/templates/report_preview.html

```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<html xmlns:th="http://www.thymeleaf.org" lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Excel Parsing Web App</title>
+</head>
+<body>
+<table>
+    <caption>Preview of the report</caption>
+    <tr>
+        <th scope="col">Location</th>
+        <th scope="col">First Name</th>
+        <th scope="col">Last Name</th>
+        <th scope="col">Event Date</th>
+        <th scope="col">Logical Device</thscope>
+    </tr>
+    <tr th:each="row : ${outputData}">
+        <td><span th:text="${row.location}"></span></td>
+        <td><span th:text="${row.firstName}"></span></td>
+        <td><span th:text="${row.lastName}"></span></td>
+        <td><span th:text="${row.eventDate}"></span></td>
+        <td><span th:text="${row.logicalDevice}"></span></td>
+    </tr>
+</table>
+</body>
+
+<form action="#" th:action="@{/download}" method="get">
+    <input type="submit" value="Download Report"/>
+</form>
+
+<form action="#" th:action="@{/restart}" method="get">
+    <input type="submit" value="Go to Home"/>
+</form>
+
+
+</html>
```

Comment:
Check this: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log#:~:text=It%20indicates%20that%20you%20do,file%20is%20not%20a%20newline.